openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/OpenApiRequestValidator.java

```diff
@@ -79,14 +79,22 @@ public ValidationResult validateRequestObject(final RequestMetaData request, Str
     private static SimpleRequest buildSimpleRequest(RequestMetaData request, String requestBody) {
         var requestBuilder = new SimpleRequest.Builder(request.getMethod(), request.getUri().getPath());
         URLEncodedUtils.parse(request.getUri(), StandardCharsets.UTF_8)
-            .forEach(p -> requestBuilder.withQueryParam(p.getName(), p.getValue()));
+            .forEach(p -> requestBuilder.withQueryParam(p.getName(), getQueryParameterValueWithOptionalDecode(p)));
         if (requestBody != null) {
             requestBuilder.withBody(requestBody);
         }
         request.getHeaders().forEach(requestBuilder::withHeader);
         return requestBuilder.build();
     }
 
+    private static String getQueryParameterValueWithOptionalDecode(NameValuePair p) {
```

Comment:
I wonder why we pass in the `NameValuePair` instead of directly the value. Why not pass in `p.getValue()` and rename the method to `nullSafeURLDecode`?